buildrunner/config_model.py

```diff
@@ -0,0 +1,154 @@
+                      mp_push_tags: Set[str],
+                      step_name: str,
+                      update_mp_push_tags: bool = True):
+        """
+        Validate push step
+
+        Args:
+            push (StepPushDict | list[str | StepPushDict] | str): Push step
+            mp_push_tags (Set[str]): Set of all tags used in multi-platform build steps
+            step_name (str): Name of the step
+            update_mp_push_tags (bool, optional): Whether to update the set of tags used in multi-platform steps.
+
+        Raises:
+            ValueError: If the config file is invalid
+        """
+        # Check for valid push section, duplicate mp tags are not allowed
+        if push is not None:
+            name = None
+            names = None
+            if isinstance(push, str):
+                name = push
+                if ":" not in name:
+                    name = f'{name}:latest'
+
+            if isinstance(push, StepPushDict):
+                names = [f"{push.repository}:{tag}" for tag in push.tags]
+
+            if names is not None:
+                for current_name in names:
+                    if current_name in mp_push_tags:
+                        # raise ValueError(f'Cannot specify duplicate tag {current_name} in build step {step_name}')
+                        raise ValueError(f'Cannot specify duplicate tag {current_name} in build step {step_name}')
+
+            if name is not None and name in mp_push_tags:
+                # raise ValueError(f'Cannot specify duplicate tag {name} in build step {step_name}')
+                raise ValueError(f'Cannot specify duplicate tag {name} in build step {step_name}')
+
+            if update_mp_push_tags and names is not None:
+                mp_push_tags.update(names)
+
+            if update_mp_push_tags and name is not None:
+                mp_push_tags.add(name)
+
+    def validate_multi_platform_build(self, mp_push_tags: Set[str]):
+        """
+        Validate multi-platform build steps
+
+        Args:
+            mp_push_tags (Set[str]): Set of all tags used in multi-platform build steps
+
+        Raises:
+            ValueError | pydantic.ValidationError: If the config file is invalid
+        """
+        # Iterate through each step
+        for step_name, step in self.steps.items():
+            if step.is_multi_platform():
+                if step.build.platform is not None:
+                    raise ValueError(f'Cannot specify both platform ({step.build.platform}) and '
+                                     f'platforms ({step.build.platforms}) in build step {step_name}')
+
+                if not isinstance(step.build.platforms, list):
+                    raise ValueError(f'platforms must be a list in build step {step_name}')
+
+                # Check for valid push section, duplicate mp tags are not allowed
+                self.validate_push(step.push, mp_push_tags, step_name)
+
+    def validate(self):
```

Comment:
I would get rid of the constructor (use the BaseModel's default constructor) and then make this a model validator or something similar: https://docs.pydantic.dev/latest/usage/validators/#model-validators.

Reply:
done